extension.js

```diff
@@ -42,6 +43,8 @@ function init() {
         watchDog.destroy();
     };
     /* eslint-enable no-undef */
+    const settings = SettingsManager.getDefaultGSettings();
+    settings.reset('recent-icons'); // clear in preparation of new collection
```

Comment:
As said in inline comment, I'd prefer to have another way to handle this.

One method would be using a GAction for example...